### PR TITLE
Fix: vault auth check

### LINKED ISSUE #387: Access control not implemented for all functions
Access control is not implemented for all functions. We need to check to make sure people are acting on vaults they own if not vault 0 (vault 0 is valid for those functions that can create a new vault).

### PR DESCRIPTION
**Pull request by haythem96**
# Fix: vault auth check

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #387 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

### CODE REVIEW

**Comment by aleone** [comment]
I think this function is fine, not sure where I saw the issue now.  I think the original implementation is fine, isnt it?

We don't ever need id 0 for rebalanceVaultNft right, so can go back to original (pre this commit)?

**Comment by haythem96** [comment]
At a first look I wasn't sure where the issue is, as all function have the correct auth check.
But for `rebalanceVaultNft()` isn't vaultId = 0 needed if user want to use `RebalanceVaultNftType.MintNewLp` action in a new vault ?

File: packages/hardhat/contracts/periphery/ControllerHelper.sol [hunk 1]
```diff
@@ -574,6 +571,13 @@ contract ControllerHelper is UniswapControllerHelper, EulerControllerHelper, IER
                 (uint256, ControllerHelperDataType.RebalanceVaultNftParams[])
             );
 
+            // check ownership
```
Comment (alpinechicken) [inline packages/hardhat/contracts/periphery/ControllerHelper.sol:574]: nice I think this fixes my problem too

**Review by alpinechicken — APPROVED** [review]
lgtm

**Review by alpinechicken — APPROVED** [review]
lgtm

**Comment by aleone** [comment]
> At a first look I wasn't sure where the issue is, as all function have the correct auth check. But for `rebalanceVaultNft()` isn't vaultId = 0 needed if user want to use `RebalanceVaultNftType.MintNewLp` action in a new vault ?

I don't really see that people will/should use this function for new vaults (we have others), but don't see a big issue including it to let people use it for new vaults.  

However, I think we should do the check in the external function vs in the callback, like it originally was, and you can add the if vaultId>0 check to what was there before. Any reason you moved it?

**Comment by haythem96** [comment]
Not needed

**Pull request closed without merging**